### PR TITLE
#167733815 Reset Password Endpoint

### PR DESCRIPTION
**Pull request by fegzycole**
#### What does this PR do?
Creates a Password Reset endpoint for a registered user to reset their password
#### Description of Task to be completed?
Install Nodemailer package
Create an endpoint to send the password reset link to the registered user
Create an endpoint to reset the user's password
#### How should this be manually tested?
The endpoint cdan be tested on api/v1/auth/sendMail to send the mail and finally api/v1/auth/resetPassword
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[https://www.pivotaltracker.com/story/show/167733815](https://www.pivotaltracker.com/story/show/167733815)
#### Screenshots (if appropriate)
N/A
#### Questions:
N/A

### CODE REVIEW

**Review by evansinho — APPROVED** [review]

**Review by codecell — APPROVED** [review]
Nice,

File: package.json [hunk 1]
```diff
@@ -19,6 +19,7 @@
   "license": "MIT",
   "dependencies": {
     "@babel/polyfill": "^7.4.4",
+    "@sendgrid/mail": "^6.4.0",
```
Comment (Oluwafayokemi) [inline package.json]: Why do you have sendgrid and node mailer at the same time, can you stick to using one of it?

File: src/middlewares/Validations/passwordValidator.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+ * @param {Object} req
+ * @param {Object} res
+ * @param {function} next
+ * @returns {(function|Object)} function next() or an error response object
+ * @description Handles validation for the password reset endpoint
+ */
+const checkForValidPassword = (req, res, next) => {
+  const { query } = req.query;
+  const { password, confirmPassword } = req.body;
+
```
Comment (Oluwafayokemi) [inline src/middlewares/Validations/passwordValidator.js:15]: This password validator, don't you think you should also add it to the sign up route. If you all really want to validate password in your app

Comment (fegzycole) [reply src/middlewares/Validations/passwordValidator.js:15]: @girlking is using a validation package for the signup route, But we can find a way to work with it.

File: package.json (hunk 1 above)
Comment (fegzycole) [reply package.json]: fixed

File: src/middlewares/Validations/passwordValidator.js (hunk 2 above)
Comment (Oluwafayokemi) [reply src/middlewares/Validations/passwordValidator.js:15]: Yes she is but she is not validating all this for the user to sign up,  We can look into this later. Well done

**Pull request merged**